apk upload to fdroid need change for new cgeo-contact addon (fix #1)

bekuno opened the pull request:
fixes #1

bekuno committed:
apk upload to fdroid need change for new cgeo-contact addon (fix #1)

fixes #1

Lineflyer commented:
If I understand correctly this will afterwards require that the contacts APK needs to added to every new release on Github, does it?

For me OK, but its a manual step, so it might be forgotten or the file name might differ each time. 

Maybe we should move contacts to a dedicated repo?

bekuno commented:
With this code - yes.
What about using separate tags for the contacts plugin?
Then we can browse 
  https://api.github.com/repos/cgeo/cgeo/tags
and filter the latest for "market_*" or "contacts_*".

Lineflyer commented:
I am not sure how it can be done to browse and find.

Important for me is, that https://github.com/cgeo/cgeo/releases/latest is used on our website as target for apk download. Technically this always points to the newest tag, which gives a problem if we the newest tag is for contacts.

Maybe best is to store all relevant apk on download.cgeo.org or any other dedicated location.

Lineflyer commented:
Another thought:

How about just keeping it hardcoded for now in the script. 
Most probably we wont have any new planned release of the plugin and mid term target would be to reintegrate anyway.

bekuno committed:
set cgeo contacts back to a direct path

bekuno commented:
Ok, then lets for now with the dedicated plugin path.
We need the extra row 
  release=$(echo $apk|grep 'https\S*cgeo-release-\S*apk' -o)
because "latest" contains currently also the entry for the plugin.

Lineflyer commented:
> release=$(echo $apk|grep 'https\Scgeo-release-\Sapk' -o)
> because "latest" contains currently also the entry for the plugin.

OK, this means when uploading new APK releases of the main app to Github the file name should contain "cgeo-release" then?

bekuno committed:
delete unnessary "-"

All latest cgeo apk packages had the name 
  cgeo-release.apk
or
  cgeo-release_XXXXXXXX.apk
(XX is the date)

In 
  https\S*cgeo-release\S*apk 
we find with \S* all non whitespace symbols. eg. current
  https://github.com/cgeo/cgeo/releases/download/market_20191111/cgeo-release.apk
